components/Switch.tsx

```diff
@@ -0,0 +1,401 @@
+    ISubmittableResult
+  > | null>(null);
+  const [tokenSymbol, setTokenSymbol] = useState<string | null>(null);
+
+  const filteredLocations = useMemo(
+    () =>
+      snowbridgeEnvironment.locations
+        .filter((x) => x.type !== "ethereum")
+        .filter((x) => x.name !== "Muse"),
+    [snowbridgeEnvironment],
+  );
+
+  const initialDestination = useMemo(
+    () => filteredLocations.find((v) => v.id === "KILT"),
+    [filteredLocations],
+  );
+
+  const form: UseFormReturn<FormDataSwitch> = useForm<
+    z.infer<typeof formSchemaSwitch>
+  >({
+    resolver: zodResolver(formSchemaSwitch),
+    defaultValues: {
+      source: filteredLocations.find((v) => v.id === "assethub"),
+      // to do: handle this better
+      destination: initialDestination ?? "",
+      token: initialDestination?.erc20tokensReceivable[0].id,
+      beneficiary: polkadotAccount?.address ?? "",
+      sourceAccount: polkadotAccount?.address ?? "",
+      amount: "0.0",
+    },
+  });
+
+  const source = form.watch("source");
+  const destination = form.watch("destination");
+  const sourceAccount = form.watch("sourceAccount");
+  const beneficiary = form.watch("beneficiary");
+  const amount = form.watch("amount");
+  const token = form.watch("token");
+
+  const beneficiaries: AccountInfo[] = useMemo(
+    () =>
+      polkadotAccounts?.map((x) => ({
+        key: x.address,
+        name: x.name || "",
+        type: destination.type,
+      })) || [],
+    [polkadotAccounts, destination.type],
+  );
+
+  const handleTransaction = useCallback(async () => {
+    if (
+      !context ||
+      !beneficiary ||
+      !source ||
+      !destination ||
+      !sourceAccount ||
+      !token
+    ) {
+      return;
+    }
+
+    const amountInSmallestUnit = parseUnits(amount, source.paraInfo?.decimals);
+    if (!amountInSmallestUnit) {
+      return;
+    }
+    const sendTransaction = async (
+      transaction: SubmittableExtrinsic<"promise", ISubmittableResult>,
+      transactionFee: string,
+    ) => {
+      setTransaction(transaction);
+      setFeeDisplay(transactionFee);
+    };
+
+    if (source.id === "assethub") {
+      if (destination.id === "assethub") {
+        return;
+      }
+      await submitAssetHubToParachainTransfer({
+        context,
+        beneficiary,
+        source,
+        destination,
+        amount: amountInSmallestUnit,
+        sourceAccount,
+        setError,
+        setBusyMessage,
+        sendTransaction,
+      });
+    } else {
+      const { pallet } = parachainConfigs[source.name];
+
+      submitParachainToAssetHubTransfer({
+        context,
+        beneficiary,
+        source,
+        amount: amountInSmallestUnit,
+        pallet,
+        sourceAccount,
+        setError,
+        setBusyMessage,
+        sendTransaction,
+      });
+    }
+  }, [context, beneficiary, source, destination, sourceAccount, token, amount]);
+
+  useEffect(() => {
+    handleTransaction();
+  }, [handleTransaction]);
```

Comment:
```suggestion
  useEffect(() => {
    const timeout = setTimeout(handleTransaction, 1000);
    return () => clearTimeout(timeout);
  }, [handleTransaction]);
```
Debouncing could also be introduced here by adding a delay after which the extrinsic creation and fee estimation is triggered. Because the timeout is reset whenever handleTransaction changes (which changes whenever one of its dependencies change) the extrinsic is only updated if there are no input changes for 1000ms